### PR TITLE
add --dry-run option to "oc adm groups new"

### PR DESCRIPTION
**Pull request by juanvallejo**
Adds a --dry-run option to `oc adm groups new`
Fixes https://github.com/openshift/origin/issues/14807

cc @soltysh

### CODE REVIEW

File: test/cmd/admin.sh [hunk 1]
```diff
@@ -117,6 +117,11 @@ echo "certs: ok"
 os::test::junit::declare_suite_end
 
 os::test::junit::declare_suite_start "cmd/admin/groups"
+# test --dry-run flag for this command
+os::cmd::expect_success_and_text 'oc adm groups new mygroup --dry-run' 'group "mygroup" created (dry run)'
```
Comment (enj) [inline test/cmd/admin.sh]: Need to escape parentheses `\(dry run\)`

**Comment by soltysh** [comment]
@juanvallejo do want to land this before we get there with our re-factors or we'll pick it up during those?

**Comment by juanvallejo** [comment]
@soltysh 
> do want to land this before we get there with our re-factors or we'll pick it up during those?

I think we can land it now, will rebase

**Comment by juanvallejo** [comment]
@soltysh rebased

**Comment by soltysh** [comment]
The error:
```bash
test/integration/groups_test.go:193:10: cannot use func literal (type func("github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/runtime".Object, "io".Writer) error) as type printers.ResourcePrinter in field value:
	func("github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/runtime".Object, "io".Writer) error does not implement printers.ResourcePrinter (missing PrintObj method)
``` 
looks relevant.

File: pkg/oc/cli/admin/groups/new/new.go [hunk 2]
```diff
@@ -3,20 +3,18 @@ package new
...
 	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
-	kprinters "k8s.io/kubernetes/pkg/printers"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/printers"
 
-	userapi "github.com/openshift/origin/pkg/user/apis/user"
-	userclientinternal "github.com/openshift/origin/pkg/user/generated/internalclientset"
-	usertypedclient "github.com/openshift/origin/pkg/user/generated/internalclientset/typed/user/internalversion"
+	userapiv1 "github.com/openshift/api/user/v1"
```
Comment (soltysh) [inline pkg/oc/cli/admin/groups/new/new.go]: @juanvallejo yeah, we'll need to sweep them afterwards, I want to get these pulls in asap.

**Pull request merged**